pkg/engine/jmespath/arithmetic.go

```diff
@@ -51,33 +51,118 @@ func ParseArithemticOperands(arguments []interface{}, operator string) (Operand,
+			var sc Scalar
+			sc.float64 = tmp.Float()
+			op[i] = sc
+		} else if tmp, err = validateArg(divide, arguments, i, reflect.String); err == nil {
+			if q, err := resource.ParseQuantity(tmp.String()); err == nil {
+				op[i] = Quantity{Quantity: q}
+			} else if d, err := time.ParseDuration(tmp.String()); err == nil {
+				op[i] = Duration{Duration: d}
+			}
+		}
+    }
+    return op, nil
+}
+
 // Quantity +|- Quantity          -> Quantity
 // Quantity +|- Duration|Scalar   -> error
 // Duration +|- Duration          -> Duration
 // Duration +|- Quantity|Scalar   -> error
 // Scalar   +|- Scalar            -> Scalar
 // Scalar   +|- Quantity|Duration -> error
 
-func (op1 Quantity) Add(op2 interface{}) (interface{}, error) {
+
+func addScalars(scalars []Operand) (interface{}, error) {
+	if len(scalars) == 0 {
+		return nil, fmt.Errorf("empty array")
+	}
+
+	result := scalars[0]
+	for i := 1; i < len(scalars); i++ {
+	
+		res, err := result.Add(scalars[i], true)
+		
+		if err != nil {
+			return nil, err
+		}
+		result = Scalar{float64: res.(float64)}
+	}
+
+	return result.(Scalar).float64, nil
+
+}
+
+func addQuantities(quantities []Operand)(interface{}, error){
+	if len(quantities) == 0 {
+        return Quantity{}, fmt.Errorf("empty array")
+    }
+
+    result := quantities[0].(Quantity)
+    for i := 1; i < len(quantities); i++ {
+        res, err := result.Add(quantities[i], false)
+        if err != nil {
+            return Quantity{}, err
+        }
+        result = res.(Quantity)
+    }
+	return result.String(), nil
+}
+
+func addDurations(durations []Operand)(interface{}, error){
+	if len(durations) == 0 {
+		return "", fmt.Errorf("empty array of durations")
+	}
+
+	// Start with the first duration in the array
+	result := durations[0]
+
+	// Add the remaining durations in the array to the result using the `Add` method
+	for i := 1; i < len(durations); i++ {
+		value, err := result.Add(durations[i], false)
+		if err != nil {
+			return "", err
+		}
+		result = Duration{value.(time.Duration)}
+	}
+
+	return result.(Duration).String(), nil
+}
+
+func (op1 Quantity) Add(op2 interface{}, check bool) (interface{}, error) {
 	switch v := op2.(type) {
 	case Quantity:
 		op1.Quantity.Add(v.Quantity)
-		return op1.String(), nil
+		if(check){
+		 return op1.String(),nil
+	}else
+	{
+		return op1, nil
+	}
 	default:
 		return nil, formatError(typeMismatchError, add)
 	}
 }
 
-func (op1 Duration) Add(op2 interface{}) (interface{}, error) {
+func (op1 Duration) Add(op2 interface{}, check bool) (interface{}, error) {
 	switch v := op2.(type) {
 	case Duration:
-		return (op1.Duration + v.Duration).String(), nil
+		if(check){
+			return (op1.Duration + v.Duration).String(), nil
+		}else{
+			return (op1.Duration + v.Duration), nil
+		}
+		
 	default:
 		return nil, formatError(typeMismatchError, add)
 	}
 }
 
-func (op1 Scalar) Add(op2 interface{}) (interface{}, error) {
+func (op1 Scalar) Add(op2 interface{}, check bool) (interface{}, error) {
```

Comment:
What is this new `bool` arg doing ?

Reply:
So, what I actually did is I used Add filter to implement the sum filter, since the Add filter adds two entities, I have added the 'check' to check whether the Add function is called for two entities, or for an array of quantities. Because for array, the return format is different and for two entities the return format is different.

Reply:
See my comment below, it would be a lot simpler, WDYT ?